### PR TITLE
[RLLib] MBMPO Fixes

### PR DESCRIPTION
**Pull request by michaelzhiluo**
Fixes the Performance Bug for MBMPO; it was a combination of initial random exploration and bug in MBMPO vectorized environment

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by michaelzhiluo** [comment]
MBMPO Experiments here: https://github.com/ray-project/rl-experiments

File: rllib/agents/mbmpo/utils.py [hunk 1]
```diff
@@ -66,3 +77,50 @@ def discount_cumsum(x, discount):
         """
     return scipy.signal.lfilter(
         [1], [1, float(-discount)], x[::-1], axis=0)[::-1]
+
+
+class MBMPOExploration(Exploration):
+    """An exploration that simply samples from a distribution.
```
Comment (sven1977) [inline rllib/agents/mbmpo/utils.py:83]: We should simply add the n-initial-random timesteps option to the existing StochasticSampling Exploration class (similar to GaussianNoise and OrnsteinUhlenbeck). Then all algos that currently use StochasticSampling could benefit from this as well. Or is there another reason, why MBMPO Exploration is special?

File: rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml [hunk 2]
```diff
@@ -1,11 +1,12 @@
-halfcheetah-mb-mpo:
-    env: ray.rllib.examples.env.halfcheetah.HalfCheetahWrapper
+halfcheetah-mbmpo:
```
Comment (sven1977) [inline rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: It would really be nice to have a small learning test case for MBMPO in our regression suite. Could we do the following?
- wrap a very simple cont. action env? Like some repeat_after_me env (where we would be able to write a reward function).
- Add a MBMPO regression test to BUILD?

**Review by sven1977 — CHANGES_REQUESTED** [review]
Great stuff. Just 2 comments, main one being: We need some regression learning test for this.

File: rllib/agents/mbmpo/utils.py (hunk 1 above)
Comment (michaelzhiluo) [reply rllib/agents/mbmpo/utils.py:83]: It is because only the local worker (not remote workers) needs the random sampling for action space!

File: rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml (hunk 2 above)
Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: One thing to keep in mind is that each gradient step in MBMPO does take a while (2-3 second), after all, higher order gradient is caculated in MAML. In addition, one needs to go through many epochs of training data for the TD models before even going through MAML.

Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: Even if it is a simple env, it might take a while... I'm wondering what's the max time allocated for regression tests?

Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: @sven1977

File: rllib/agents/maml/maml_torch_policy.py [hunk 3]
```diff
@@ -230,10 +229,12 @@ def __init__(self,
...
 
         self.inner_kl_loss = torch.mean(
-            torch.stack(
-                [a * b for a, b in zip(self.cur_kl_coeff, mean_inner_kl)]))
+            torch.stack([
+                a * b for a, b in zip(self.cur_kl_coeff, self.mean_inner_kl)
+            ]))
         self.loss = torch.mean(torch.stack(ppo_obj)) + self.inner_kl_loss
-        print("Meta-Loss: ", self.loss, ", Inner KL:", self.inner_kl_loss)
+        print("Meta-Loss: ", self.loss.item(), ", Inner KL:",
```
Comment (sven1977) [inline rllib/agents/maml/maml_torch_policy.py:236]: Let's remove this here. We shouldn't print out any stats during training.

File: rllib/agents/mbmpo/utils.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/mbmpo/utils.py:83]: Got it. Makes sense. Thanks for clarifying.

File: rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml (hunk 2 above)
Comment (sven1977) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: Some take a couple of minutes (I think DDPG or PPO Pendulum are the slowest ones), which is totally fine. Would anything (very simple) be possible to learn in let's say 5min?

**Review by sven1977 — APPROVED** [review]
Looks great! Could you remove the print of the loss, though, and pull master again? Then we can merge this.

File: rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml (hunk 2 above)
Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: Not sure XD I'll try it out on pendulum and lyk how long it takes to get past -800 rew

Comment (michaelzhiluo) [reply rllib/tuned_examples/mbmpo/halfcheetah-mbmpo.yaml:1]: Aight it does not seem possible, b/c training models to a decent level first to sample from takes a while....

**Comment by michaelzhiluo** [comment]
@sven1977 Aight, updated to master and deleted comment. I think its good to go!
